chrysom/basicClient.go

```diff
@@ -86,6 +93,31 @@ const (
 // Items is a slice of model.Item(s) .
 type Items []model.Item
 
+// GetLogger returns a logger from the given context.
+type GetLogger func(context.Context) *zap.Logger
+
+// SetLogger embeds the `Listener.logger` in outgoing request contexts for `Listener.Update` calls.
+type SetLogger func(context.Context, *zap.Logger) context.Context
```

Comment:
Introduced get/set logger types to simplify fx setup (they're only used for fx.in structs).